sdk/identity/azure-identity/src/main/java/com/azure/identity/ManagedIdentityServiceCredential.java

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.identity.implementation.IdentityClient;
+import reactor.core.publisher.Mono;
+
+/**
+ * The Managed Service Identity credential.
+ */
+abstract class ManagedIdentityServiceCredential {
+    private final String clientId;
+    private final String environment;
+    final IdentityClient identityClient;
+
+    /**
+     * Creates an instance of ManagedIdentityServiceCredential.
+     * @param clientId the client id of user assigned or system assigned identity
+     * @param identityClient the identity client to acquire a token with.
+     * @param environment The service environment of the credential.
+     */
+    ManagedIdentityServiceCredential(String clientId, IdentityClient identityClient, String environment) {
+        this.identityClient = identityClient;
+        this.clientId = clientId;
+        this.environment = environment;
```

Comment:
Can this be converted into an enum where all the environments are listed?

Reply:
Yeah since this is not public API, internally it can be introduced as an impl detaiI.
But, not critically needed currently, as this info is used only in the log message currently.
I'll introduce it in a follow up code refactor PR.